### PR TITLE
Make `get_special_tokens_mask` consider all tokens

### LINKED ISSUE #11155: [BUG] padding tokens are also masked in DataCollatorForLanguageModeling
## Environment info

- `transformers` version: 4.3.2
- Platform: Linux
- Python version: 3.6
- PyTorch version (GPU?): 1.7.1 GPU
- Tensorflow version (GPU?): N/A
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: Sagemaker distributed data parallel

### Who can help 
@sgugger 

## Information

Model I am using (Bert, XLNet ...): All models that use DataCollatorForLanguageModeling.

The bug is introduced in this [PR](https://github.com/huggingface/transformers/pull/8308). 

3 lines (241-243) are removed by mistake from this [line](https://github.com/huggingface/transformers/pull/8308/commits/74b3d7abce96c79bf8c35517857b4032b3d85a21#diff-046566f2b40a246c7d533457cd7f6f07830516da845b904086f36b3cfe0d5965L241).

Now padding tokens are also masked in MLM.

The problem arises when using:
* [x] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

```
from transformers import DataCollatorForLanguageModeling
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained('albert-base-v2')
data_collator = DataCollatorForLanguageModeling(
                tokenizer=tokenizer, mlm=True, mlm_probability=0.15
            )
tok = tokenizer('hello how are you!', add_special_tokens=True, truncation=True, max_length=256, padding='max_length')
data_collator([tok['input_ids']])
```

From the output you can easily see that the padding tokens are masked. Add back the three removed lines fix this bug.

## Expected behavior

padding token is not supposed to be mask-able in MLM.

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

As discovered via #11155, some tokenizers do not return the proper special tokens mask in the `get_special_tokens_mask` when `already_has_special_tokens=True` because they only check for the CLS and SEP tokens. This PR fixes that by delegating to the superclass the call when `already_has_special_tokens=True` (the generic method checks for all special tokens).

It also seems from the error message [here](https://github.com/huggingface/transformers/blob/b9b60c1630f63b54b10380ef8bf30ec323985553/src/transformers/tokenization_utils_base.py#L3091) that the `get_special_tokens_mask` method is not supposed to be implemented for fast tokenziers when `already_has_special_tokens=True`, so this PR removes this method from the fast tokenizers where it exists, except for a select few that have a different implementation.

Fixes #11155

### CODE REVIEW

**Comment by woong97** [comment]
Thanks a lot for your super-fast feedback! Your projects are big inspiration to me. Thank you.

**Review by LysandreJik — APPROVED** [review]
Yes, this LGTM! Let's keep a close eye at the slow integration tests to make sure that nothing breaks from this.
